core/DataAccess/LogQueryBuilder/JoinGenerator.php

```diff
@@ -190,8 +191,20 @@ public function generate()
                     continue;
                 }
 
+                $joinName = 'LEFT JOIN';
+                if ($i > 0
+                    && $this->tables[$i - 1]
+                    && is_string($this->tables[$i - 1])
+                    && strpos($this->tables[$i - 1], LogAggregator::LOG_TABLE_SEGMENT_TEMPORARY_PREFIX) === 0) {
+                    $joinName = 'INNER JOIN';
+                    // edge case when there is eg `logtmpsegment57cd546b7203d68a41027547c4abe1a2` as first table...
+                    // then we join log_conversion for example... if we didn't use INNER JOIN we would as a result
+                    // get rows for visits even when they didn't have a conversion. Instead we only want to find rows
+                    // that have an entry in both tables when doing eg
+                    // logtmpsegment57cd546b7203d68a41027547c4abe1a2.idvisit = log_conversion.idvisit
+                }
                 // the join sql the default way
-                $this->joinString .= " LEFT JOIN $tableSql ON " . $join;
+                $this->joinString .= " $joinName $tableSql ON " . $join;
```

Comment:
Would it make this code simpler to make sure the log table segment is always last? I guess it's possible the segment table being first is a valid use case...

Reply:
The temporary log table should always be first otherwise things get slow. And for regular archives without segment there is no tmp log table as it would make things slower. Not sure if that helps?

Reply:
If it's always first, then this isn't an edge case? I was responding to the comment above.

Reply:
Yep logtemptableXYZ is always first when there is a segment applied.

Reply:
So then the comment about the edge case is incorrect?

Reply:
Do you refer to it being an edge case? I guess it's not an edge case considering that segments will be archived fairly often. Was just thinking of it as an edge case as it is only when there are segments but not for regular archive

Reply:
seems like the new normal behavior to me. when I read the comment I thought the segment table being first was a strange thing that could happen, not something that should happen if there's a segment.

Reply:
changed it 👍